Add documentation regarding specification URL and OData (#3697)
kyma-project/kyma: docs/application-connector/docs/08-03-register-manage-services.md

    ```
    curl https://gateway.kyma.local:{NODE_PORT}/{APP_NAME}/v1/metadata/services/{YOUR_SERVICE_ID} --cert {CERT_FILE_NAME}.crt --key {KEY_FILE_NAME}.key -k
    ```

> ## Register API with specification URL

Reviewer: ```suggestion
## Register API with a specification URL
```

Reviewer 2: Done!

> Application Registry allows you to pass API specification in form of specification URL.

Reviewer: ```suggestion
The Application Registry allows you to pass API specifications in a form of specification URLs.
```

Reviewer 2: Done!

> To register API with specification URL replace `api.spec` with `api.specificationUrl`. 

Reviewer: ```suggestion
To register API with specification URL, replace `api.spec` with `api.specificationUrl`. 
```

> To register API with specification URL replace `api.spec` with `api.specificationUrl`. Be aware that `api.spec` has higher priority than `api.specificationUrl`.

Reviewer 3: I would rephrase this sentence to state that if you provide both api.spec and api.specificationUrl then api.spec will be used.

Reviewer 2: Done!

> To register API with specification URL replace `api.spec` with `api.specificationUrl`. Be aware that `api.spec` has higher priority than `api.specificationUrl`.

Reviewer: ```suggestion
To register API with a specification URL, replace **api.spec** with **api.specificationUrl**. Note that **api.spec** has a higher priority than **api.specificationUrl**.
```

Reviewer: ``Be aware that `api.spec` has higher priority than `api.specificationUrl`.`` -> so what? Please provide more info.

Reviewer 2: Done!

> ```

Reviewer: What is that? Please describe the example in one sentence, for example:
`See the example of blablabla:`

Reviewer 2: That's the API part of the request body if the user wants to register their API with specification URL. 
```
See the example of the API part of the request body with specification URL:
...
>**NOTE** If both api.spec and api.specificationUrl are provided, api.spec will be used due to its higher priority.
```
Would that be better, @klaudiagrz ?

Reviewer: That's exactly what I meant :)

> 

Reviewer: ```suggestion
See the example of the API part of the request body with specification URL:
```

  "targetUrl": "https://services.odata.org/OData/OData.svc",
  "specificationUrl": "https://services.odata.org/OData/OData.svc/$metadata",
  "credentials": {
    "basic": {
      "username": "{USERNAME}",
      "password": "{PASSWORD}"
    }
}
```

> The Application Registry will fetch the specification from provided URL but it will not use any credentials, therefor the endpoint can not be secured by any authentication mechanism.

Reviewer 3: Typo "therefor"

> The Application Registry will fetch the specification from provided URL but it will not use any credentials, therefor the endpoint can not be secured by any authentication mechanism.

Reviewer: ```suggestion
The Application Registry fetches specification from the provided URL but it does not use any credentials, therefore the endpoint cannot be secured by any authentication mechanism.
```

Reviewer 2: Done!

> >**NOTE:** Fetching specification from URL is supported only for API Spec. It can not be done for Events and Documentation.

Reviewer: ```suggestion
>**NOTE:** Fetching specification from a URL is supported only for APIs. Fetching specifications for Events or documentation is not supported.
```

Reviewer 2: Done!

> ## Registering OData API

Reviewer: ```suggestion
## Register the OData API
```

Reviewer 2: Done!

> If no `api.spec` or `api.specificationUrl` are specified and `api.type` is set to `OData`, the Application Registry will try to fetch the specification from the target URL with the `$metadata` path.

Reviewer: ```suggestion
If the **api.spec** or **api.specificationUrl** parameters are not specified and the **api.type** parameter is set to `OData`, the Application Registry will try to fetch specification from the target URL with the `$metadata` path.
```

Reviewer 2: Done!

> For service with the following api:

Reviewer: ```suggestion
For example, for the service with the following API, the Application Registry will try to fetch API specification from `https://services.odata.org/OData/OData.svc/$metadata`.
```

Reviewer 2: Done!

```
"api": {
  "targetUrl": "https://services.odata.org/OData/OData.svc",
  "apiType": "OData"
  "credentials": {
    "basic": {
      "username": "{USERNAME}",
      "password": "{PASSWORD}"
    }
}
```

> Application Registry will try to fetch API Spec from `https://services.odata.org/OData/OData.svc/$metadata`.

Reviewer: Delete.

Reviewer 2: Done!
